elektronLogging/ElektronFormatter.go

```diff
@@ -0,0 +1,55 @@
+package elektronLogging
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+    "github.com/fatih/color"
+    log "github.com/sirupsen/logrus"
+)
+
+type ElektronFormatter struct {
+	TimestampFormat string
+}
+
+func (f ElektronFormatter) getColor(entry *log.Entry) *color.Color {
+    switch entry.Level {
+        case log.InfoLevel:
+            return color.New(color.FgGreen, color.Bold)
+        case log.WarnLevel:
+            return color.New(color.FgYellow, color.Bold)
+        case log.ErrorLevel:
+            return color.New(color.FgRed, color.Bold)
+        case log.FatalLevel:
+            return color.New(color.FgRed, color.Bold)
+        default:
+            return color.New(color.FgWhite, color.Bold)
+    }
+}
+func (f ElektronFormatter) Format(entry *log.Entry) ([]byte, error) {
+	var b *bytes.Buffer
+
+	if entry.Buffer != nil {
+		b = entry.Buffer
+	} else {
+		b = &bytes.Buffer{}
+	}
+
+    levelColor := f.getColor(entry)
+	level := levelColor.Sprintf("[%s]:",strings.ToUpper(entry.Level.String()))
+    message := fmt.Sprintf("%s %s ",level,entry.Time.Format(f.TimestampFormat))
+    if entry.Message != "" {
+	    message = fmt.Sprintf("%s %s %s ",level,entry.Time.Format(f.TimestampFormat), entry.Message)
+    }
+	var formattedFields []string
+	for key, value := range entry.Data {
+		formattedFields = append(formattedFields, 
+		strings.Join([]string{key, fmt.Sprintf("%s", value)}, "="))
```

Comment:
This seems really inefficient. Wouldn't value already have to be a string for this to succeed? In which case, I don't think we gain anything here by suing a sprintf.

Reply:
value is not a string here, it is of type interface{}. But to avoid using sprintf, would it be fine to use type assertion? For eg : `value.(string) `

Reply:
Peeked at how [logrus](https://github.com/sirupsen/logrus/blob/67a7fdcf741f4d5cee82cb9800994ccfd4393ad0/text_formatter.go#L315) serializes values and seems like fmt.Sprint is used if type assertion fails.

Reply:
ahh alright, I thought it was a string coming in. Alright, if it's been sanity checked then i'm good with it.